Failed test test_cumul_number_matches_vs_threshold

First time I'm observing this after running `pytest` on a OSX machine, with python3.6 and clang compiler:
```
=================================== FAILURES ===================================
____________________ test_cumul_number_matches_vs_threshold ____________________

    @hypothesis.given(candidate_pairs_2p,
>                     hypothesis.strategies.integers(
                          min_value=1,
                          max_value=100,  # Linear in this
                          ))
    def test_cumul_number_matches_vs_threshold(candidate_pairs, steps):

tests/test_stats.py:75: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/vsts/hostedtoolcache/Python/3.6.9/x64/lib/python3.6/site-packages/hypothesis/core.py:610: in execute
    result = self.test_runner(data, run)
/Users/vsts/hostedtoolcache/Python/3.6.9/x64/lib/python3.6/site-packages/hypothesis/executors.py:58: in default_new_style_executor
    return function(data)
/Users/vsts/hostedtoolcache/Python/3.6.9/x64/lib/python3.6/site-packages/hypothesis/core.py:601: in run
    return test(*args, **kwargs)
tests/test_stats.py:75: in test_cumul_number_matches_vs_threshold
    hypothesis.strategies.integers(
/Users/vsts/hostedtoolcache/Python/3.6.9/x64/lib/python3.6/site-packages/hypothesis/core.py:548: in test
    result = self.test(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

candidate_pairs = (array('d', [0.5000000000000003, 0.5000000000000001]), (array('I', [0, 0]), array('I', [1, 1])), (array('I', [0, 0]), array('I', [1, 0])))
steps = 3

    @hypothesis.given(candidate_pairs_2p,
                      hypothesis.strategies.integers(
                          min_value=1,
                          max_value=100,  # Linear in this
                          ))
    def test_cumul_number_matches_vs_threshold(candidate_pairs, steps):
        counts, thresholds = anonlink.stats.cumul_number_matches_vs_threshold(
            candidate_pairs, steps=steps)
    
        assert len(counts) == len(thresholds) == steps + 1
>       assert len(set(thresholds)) == len(thresholds)
E       assert 3 == 4
E        +  where 3 = len({0.5000000000000001, 0.5000000000000002, 0.5000000000000003})
E        +    where {0.5000000000000001, 0.5000000000000002, 0.5000000000000003} = set(array([0.5, 0.5, 0.5, 0.5]))
E        +  and   4 = len(array([0.5, 0.5, 0.5, 0.5]))

tests/test_stats.py:84: AssertionError
---------------------------------- Hypothesis ----------------------------------
Falsifying example: test_cumul_number_matches_vs_threshold(candidate_pairs=(array('d', [0.5000000000000003, 0.5000000000000001]),
 (array('I', [0, 0]), array('I', [1, 1])),
 (array('I', [0, 0]), array('I', [1, 0]))), steps=3)
```
The test was usually passing. 
Whole command:
`pytest --cov=anonlink --junitxml=testoutput.xml --cov-report=xml:coverage.xml --cov-report=html:htmlcov -W ignore::DeprecationWarning`, has run on Azure Dev-Ops.

Looking at 

```
        assert len(counts) == len(thresholds) == steps + 1
>       assert len(set(thresholds)) == len(thresholds)
E       assert 3 == 4
E        +  where 3 = len({0.5000000000000001, 0.5000000000000002, 0.5000000000000003})
E        +    where {0.5000000000000001, 0.5000000000000002, 0.5000000000000003} = set(array([0.5, 0.5, 0.5, 0.5]))
E        +  and   4 = len(array([0.5, 0.5, 0.5, 0.5]))
```

I think the assertions should be [assume](https://hypothesis.readthedocs.io/en/latest/details.html#hypothesis.assume) calls instead. @wilko77?

--- a/tests/test_stats.py
+++ b/tests/test_stats.py
@@ -81,7 +81,11 @@ def test_cumul_number_matches_vs_threshold(candidate_pairs, steps):
         candidate_pairs, steps=steps)
     
     assert len(counts) == len(thresholds) == steps + 1
-    assert len(set(thresholds)) == len(thresholds)
+    # if we have less pairs than steps, then the threshold bins are not necessarily unique. E.g: two pairs with
+    # similarities 0.5000000000000003, 0.5000000000000001, and steps=3, we end up with the thresholds
+    # [0.5000000000000001, 0.5000000000000002, 0.5000000000000001, 0.5000000000000003]
+    if len(candidate_pairs[0]) >= steps:
+        assert len(set(thresholds)) == len(thresholds)
     assert sorted(thresholds) == list(thresholds)
 
     for count, threshold in zip(counts, thresholds):

